setup/main.go

```diff
@@ -0,0 +1,250 @@
+
+		steps.ConfirmAutoExplainSetup,
+		steps.CheckUseLogBasedExplain,
+		steps.CreateLogExplainHelper,
+		steps.CheckAutoExplainAvailable,
+		steps.EnableAutoExplain,
+
+		steps.ReloadCollector,
+		steps.RestartPostgres,
+		steps.ConfigureAutoExplain,
+		steps.RunPgSleep,
+	}
+
+	var setupState state.SetupState
+	var quiet bool
+	var logFile string
+	var inputsFile string
+	var recommended bool
+	flag.StringVar(&setupState.ConfigFilename, "config", defaultConfigFile, "specify alternative path for config file")
+	flag.BoolVar(&quiet, "quiet", false, "omit verbose logging output")
+	flag.StringVar(&logFile, "log", "", "save output to log file (always includes verbose output)")
+	flag.StringVar(&inputsFile, "inputs", "", "do not prompt for user inputs and use JSON file describing answers to all setup prompts")
+	flag.BoolVar(&recommended, "recommended", false, "do not prompt for user inputs and use recommended values (the --inputs flag can override individual settings)")
+	flag.Parse()
+
+	logger := log.NewLogger()
+	if logFile == "" {
+		if !quiet {
+			logger.VerboseOutput = os.Stdout
+		}
+	} else {
+		log, err := os.Create(logFile)
+		if err != nil {
+			fmt.Printf("ERROR: could not open log file %s for writes: %s\n", logFile, err)
+			os.Exit(1)
+		}
+		defer log.Close()
+		outputBoth := io.MultiWriter(os.Stdout, log)
+		logger.StandardOutput = outputBoth
+		if quiet {
+			logger.VerboseOutput = log
+		} else {
+			logger.VerboseOutput = outputBoth
+		}
+	}
+	setupState.Logger = &logger
+
+	var inputs state.SetupInputs
+	if recommended {
+		inputs = s.RecommendedInputs
+	}
+	if inputsFile != "" {
+		inputsReader, err := os.Open(inputsFile)
+		if err != nil {
+			setupState.Log("ERROR: could not open inputs file %s: %s", logFile, err)
+			os.Exit(1)
+		}
+		inputsBytes, err := ioutil.ReadAll(inputsReader)
+		if err != nil {
+			setupState.Log("ERROR: could not open inputs file %s: %s", logFile, err)
+			os.Exit(1)
+		}
+		err = json.Unmarshal(inputsBytes, &inputs)
+		if err != nil {
+			setupState.Log("ERROR: could not parse inputs file %s: %s", logFile, err)
+			os.Exit(1)
+		}
+		inputs.Scripted = true
+		err = os.Stdin.Close()
+		if err != nil {
+			setupState.Log("ERROR: could not close stdin for scripted input: %s", err)
+			os.Exit(1)
+		}
+	}
+	setupState.Inputs = &inputs
+
+	id := os.Geteuid()
+	if id > 0 {
+		setupState.Log(`ERROR: The pganalyze collector guided setup must be run as root (or with sudo, if available)
+
+It will provide details on the process and prompt you before making any changes to the
+collector config file or your database. If you prefer, you can instead follow
+the manual collector install instructions: https://pganalyze.com/docs/install`)
+		os.Exit(1)
+	}
+
+	setupState.Log(`Welcome to the pganalyze collector guided setup!
+
+We will go through a series of steps to set up the collector to monitor your
+Postgres server. We will not make any changes to Postgres or your system
+without confirmation.
```

Comment:
I wonder if we should add a disclaimer somewhere here, that this only supports a certain kind of setup:

```
IMPORTANT: Please note that this setup only works when monitoring a self-managed system, and installing the collector directly on your database server. For other setup types, please check the installation instructions https://pganalyze.com/docs/install
```

(in particular, there are cases where you run the collector on a VM, but you monitor a remote database server, e.g. for unsupported PaaS providers - in case the user ends up in this by accident, we should be very clear on this)

Reply:
We do have checks later, but you're right, I'll add something up front.